modules/lnd/lnd_wrapper/wrapper.go

```diff
@@ -130,4 +137,30 @@ func LndDeserializeInvoice(cInvoiceStr *C.char) *C.char {
 	return C.CString(sb.String())
 }
 
+//export LndParseP2pLightningMessage
+func LndParseP2pLightningMessage(data C.ByteArray) *C.char {
+	buffer := C.GoBytes(unsafe.Pointer(data.data), data.length)
```

Comment:
Nit: If we can use `unsafe.Slice` here, the line gets slightly simpler.
```suggestion
func LndParseP2pLightningMessage(data C.ByteArray) *C.char {
	buffer := unsafe.Slice(data.data, data.length)
```

Reply:
We can't really do that because the `C.GoBytes` is the function that will convert the c bytes to go bytes[]